drivers/temperature/max31865pmb1/max31865.c

```diff
@@ -0,0 +1,361 @@
+ * @param init_param - Init parameter for descriptor
+ * @return 0 in case of success, errno errors otherwise
+ */
+int max31865_init(struct max31865_dev **device,
+		  struct max31865_init_param *init_param)
+{
+	int ret;
+	struct max31865_dev *descriptor;
+
+	descriptor = (struct max31865_dev *)calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = no_os_spi_init(&descriptor->comm_desc, &init_param->spi_init);
+
+	if (ret){
+		free(descriptor);
+		return ret;
+	}
+
+	*device = descriptor;
+
+	return 0;
+}
+
+/**
+ * @brief Remove resources allocated by the init function
+ * @param device  - max31865 descriptor
+ * @return 0 in case of success, errno errors otherwise
+ */
+int max31865_remove(struct max31865_dev *device)
+{
+	int ret;
+	if (!device)
+		return -ENODEV;
+
+	ret = no_os_spi_remove(device->comm_desc);
+	if (ret)
+		return -EINVAL;
+
+	free(device);
+
+	return 0;
+}
+
+/**
+ * @brief Read raw register value
+ * @param device - MAX31865 descriptor
+ * @param val - register value
+ * @return 0 in case of success, negative error code otherwise
+ */
+int max31865_read_raw(struct max31865_dev *device, uint8_t *val)
+{
+	if((*val & 0x7F) != *val)
+		*val &= 0x7F;
+
+	if(*val < 0x00 || *val > 0x07)
+		return ENXIO;
+
+	int ret;
+	uint8_t raw_array[1];
+
+	raw_array[0] = *val;
+
+	ret = no_os_spi_write_and_read(device->comm_desc, raw_array, 1);
+
+	if (ret)
+		return ret;
+
+	*val = raw_array[0];
+
+	return 0;
+}
+
+/**
+ * @brief Write raw register value
+ * @param device - MAX31865 descriptor
+ * @param val - register value
+ * @param val2 - data to write in register
+ * @return 0 in case of success, negative error code otherwise
+ */
+int max31865_write_raw(struct max31865_dev *device, uint8_t val, uint8_t val2)
+{
+	if((val & 0x80) != 0x80)
+		val |= 0x80;
+
+	if(val < 0x00 || val > 0x07)
+		return ENXIO;
+
+	int ret;
+	uint8_t raw_array[2];
+	raw_array[0] = val;
+	raw_array[1] = val2;
+
+	ret = no_os_spi_write_and_read(device->comm_desc, raw_array, 2);
+	if (!ret)
+		return ret;
+
+	return 0;
+}
+
+/**
+ * @brief Read the raw 8-bit FAULTSTAT register
+ * @param device MAX31865 descriptor
+ * @return The raw unsigned 8-bit FAULT status register
+ */
+uint8_t max31865_read_fault(struct max31865_dev *device)
+{
+	uint8_t faultReg = MAX31865_FAULTSTAT_REG;
+	max31865_read_raw(device, &faultReg);
+
+	return faultReg;
+}
+
+/**
+ * @brief Clear all faults in FAULTSTAT
+ * @param device MAX31865 descriptor
+ */
+void max31865_clear_fault(struct max31865_dev *device)
+{
+	uint8_t faultReg = 0x00;
+	max31865_read_raw(device, &faultReg);
+	faultReg &= ~0x2C;
+	faultReg |= 0x02;
+	uint8_t clrReg = 0x00;
+	max31865_write_raw(device, clrReg, faultReg);
+}
+
+/**
+ * @brief Enable the bias voltage on the RTD sensor
+ * @param device MAX31865 descriptor
+ * @param b If true bias is enabled, else disabled
+ */
+void max31865_enable_bias(struct max31865_dev *device, bool b)
```

Comment:
same here, check the return codes for the read/write functions and change the function type to `int`. Also i'f use a better naming for variable `b` -> `bias_en`